src/site/content/en/fast/codelab-optimize-third-party-javascript/index.md

```diff
@@ -0,0 +1,248 @@
+
+However, the D3 script is included in the page's `<head>`, which blocks the parsing of the rest document: 
+
+![Screenshot of index.html with highlighted script tag in the head.](index.png)
+
+Two magic words that can unblock the parser are: 
+
+* [`async`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-async) ensures that scripts download in the background and execute **at the first opportunity** after they finish downloading.
+
+* [`defer`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-defer) ensures that scripts download in the background and execute **after parsing** is completely finished.
+
+Since this chart is not really critical to the overall page and will most likely be below the fold, use `defer` to make sure there’s no parser blocking. 
+
+### Step 1: Load the script asynchronously with the `defer` attribute
+
+{% Instruction 'remix' %}
+
+On line 17 in `index.html`, add the `defer` attribute to the `<script>` element:
+
+
+```html
+<script src="https://d3js.org/d3.v3.min.js" defer></script>
+```
+
+### Step 2: Ensure the correct order of operations
+
+Now that D3 is deferred, `script.js` will run before D3 is ready, resulting in an error.
+
+Scripts with the `defer` attribute execute in the order in which they were specified. To ensure `script.js` gets executed after D3 is ready, add `defer` to it and move it up to the `<head>` of the document, right after the D3 `<script>` element. Now it no longer blocks the parser, and the download starts sooner.
+
+```html
+<script src="https://d3js.org/d3.v3.min.js" defer></script>
+<script src="./script.js" defer></script>
+```
+
+## Lazy-load third-party resources
+
+All resources that are below the fold are good candidates for [lazy-loading](/efficiently-load-third-party-javascript/#lazy-load-third-party-resources). 
+
+The sample app has a YouTube video embedded in an [iframe](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe). To check out how many requests the page makes and which come from the embedded YouTube iframe:
+
+{% Instruction 'devtools-network', 'ol' %}
+{% Instruction 'disable-cache', 'ol' %}
+1. Select *Fast 3G* in the **Throttling** dropdown menu.
+{% Instruction 'reload-page', 'ol' %}
+
+![Screenshot of DevTools Network panel.](network.png)
+
+The **Network** panel reveals that the page made a total of 28 requests and loaded almost 1 MB of compressed resources.
+
+To identify the requests that the YouTube `iframe` made, look for the video ID `6lfaiXM6waw` in the **Initiator** column. To group together all the requests by domain:
```

Comment:
Domain sorting is just a short-cut in this case because it's easy to connect the dots: there's a bunch of Google domains -> ofc that YouTube called them. 

Initiator column is the only way to follow the trail of propagated calls: 

* `initiator:` 6lfaiXM6waw (youtube.com) loads `resource:` base.js (also youtube.com)
* `initiator:` base.js (youtube.com) loads `resource:` ads.js (from doubleclick.google.com)
* etc.

If it was any other third-party embed the grouping would be not be so clear when sorted by domain.

I understand now that this not at all obvious from the codelab, but I really don't want to drag out that step.